### PR TITLE
Add New Board: Raytac MDBT50Q Demo Board 40

### PR DESCRIPTION
**Pull request by markbirss**
* adding Raytac MDBT50Q Demo Board 40 files

### CODE REVIEW

File: src/boards/raytac_mdbt50q_db_40/board.h [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+#define BLEDIS_MANUFACTURER     "Raytac"
+#define BLEDIS_MODEL            "MDBT50Q-DB-40"
+
+//--------------------------------------------------------------------+
+// USB
+//--------------------------------------------------------------------+
+
+#define USB_DESC_VID           0x239A
+#define USB_DESC_UF2_PID       0x010C
+#define USB_DESC_CDC_ONLY_PID  0x010C
```
Comment (hathach) [inline src/boards/raytac_mdbt50q_db_40/board.h:67]: it is required by this repo that each board must have its own unique VID/PID. I double check, but this PID isn't assigned by Adafruit to any boards yet. Please try to get one from PID donor.

**Review by hathach — CHANGES_REQUESTED** [review]
VID/PID must be uniquely assigned. check out review comment for detail

**Comment by markbirss** [comment]
Ok, i understand your comment, but what is "PID donor" ?

**Comment by hathach** [comment]
> Ok, i understand your comment, but what is "PID donor" ?

I mean you could ask from one of the manufacturer to get an PID assigned to this board.

**Comment by markbirss** [comment]
Ok, will ask Raytac about it

**Comment by markbirss** [comment]
The device was loaded with this when i reeived it
ID 1915:521a Nordic Semiconductor ASA nRF52 USB CDC BLE Demo

**Comment by markbirss** [comment]
@hathach  Ok, i just found this github issue about the same thing for circuitpython for the same device

https://github.com/adafruit/circuitpython/issues/3128#issuecomment-655077754

This suggests I must use 

PID 0x00BB  # bootloader ?

**Comment by hathach** [comment]
@markbirss it is the same board, so yes, we should/must use that PID. Please update your PR

**Comment by markbirss** [comment]
> 0x00BB

Sure, i have made the required change

**Review by hathach — CHANGES_REQUESTED** [review]
board-id is not correct, it must follow uf2 convention.

File: src/boards/raytac_mdbt50q_db_40/board.h [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+//--------------------------------------------------------------------+
+
+#define USB_DESC_VID           0x239A
+#define USB_DESC_UF2_PID       0x00BB
+#define USB_DESC_CDC_ONLY_PID  0x00BB
+
+//------------- UF2 -------------//
+#define UF2_PRODUCT_NAME        "Raytac MDBT50Q Demo Board 40"
+#define UF2_VOLUME_LABEL        "MDBT50QBOOT"
+#define UF2_BOARD_ID            "RAYTAC-MDBT50Q-DB-40"
```
[the remark this answers is not in the document]
Comment (markbirss) [reply src/boards/raytac_mdbt50q_db_40/board.h:72]: @hathach 
Im not sure I understand your comment ?

Looking at the other already defined Raytec Board, what is the board-id you referring to ?

https://github.com/adafruit/Adafruit_nRF52_Bootloader/blob/master/src/boards/raytac_mdbt50q_rx/board.h
```
//------------- UF2 -------------//
#define UF2_PRODUCT_NAME        "Raytac MDBT50Q-RX"
#define UF2_VOLUME_LABEL        "MDBT50QBOOT"
#define UF2_BOARD_ID            "nRF52840-MDBT50Q_RX-verD"
#define UF2_INDEX_URL           "https://www.adafruit.com/product/5199"
```

[the remark this answers is not in the document]
Comment (hathach) [reply src/boards/raytac_mdbt50q_db_40/board.h:72]: board-id must follow format dictate by uf2 spec https://github.com/microsoft/uf2#files-exposed-by-bootloaders

[the remark this answers is not in the document]
Comment (markbirss) [reply src/boards/raytac_mdbt50q_db_40/board.h]: @hathach  Ok, for your review, made changes to UF2_BOARD_ID

**Review by hathach — COMMENTED** [review]
everything look good, only one last thing: you need add the board to ci list so that its binaries is generated when released https://github.com/adafruit/Adafruit_nRF52_Bootloader/blob/master/.github/workflows/githubci.yml

**Comment by markbirss** [comment]
@hathach  Ok, here is the PR for that
https://github.com/adafruit/Adafruit_nRF52_Bootloader/pull/245

**Review by hathach — APPROVED** [review]

**Pull request merged**